Extensions/timestamps.js

```diff
@@ -230,6 +265,89 @@ XKit.extensions.timestamps = new Object({
 		.catch(() => this.show_failed(date_element));
 	},
 
+	react_add_timestamps: function() {
+		var posts = $(XKit.extensions.timestamps.posts_class).not(".xkit_timestamps");
+
+		if (!posts || posts.length === 0) {
+			return;
+		}
+
+		XKit.extensions.timestamps.check_quota();
+
+		posts.each(function() {
+			var post = $(this);
+
+			post.addClass("xkit_timestamps");
+
+			// What is the react class for .fan_mail?
+//			if (post.hasClass("fan_mail")) {
+//				return;
+//			}
+
+			// What is the react id for #new_post?
+//			if (post.attr('id') === "new_post" || post.find(XKit.extensions.timestamps.private_posts_class).length > 0)
+			if (post.find(XKit.extensions.timestamps.private_posts_class).length > 0) {
+				return;
+			}
+
+			var post_id = $(this).parents("[data-id]").attr("data-id");
+
+			var xtimestamp_class = "xtimestamp";
+			if (XKit.extensions.timestamps.in_search && !$("#search_posts").hasClass("posts_view_list")) {
+				xtimestamp_class = "xtimestamp-in-search";
+			}
+
+			var xtimestamp_html = `<div class="xkit_timestamp_${post_id} ${xtimestamp_class} xtimestamp_loading">&nbsp;</div>`;
+			post.find(XKit.extensions.timestamps.meatballs_class).prepend(xtimestamp_html);
+
+			var note = $(".xkit_timestamp_" + post_id);
+			XKit.extensions.timestamps.react_fetch_timestamp(post_id, note);
+		});
+	},
+
+	react_add_reblog_timestamps: function() {
+		var selector = XKit.extensions.timestamps.reblogs_class;
+		if (XKit.extensions.timestamps.preferences.reblogs.value === "op") {
+			selector += ` > ${XKit.extensions.timestamps.reblog_op_class}`;
+		}
+
+		$(selector).not(".xkit_timestamps")
+		.addClass("xkit_timestamps")
+		.each(function() {
+			try {
+				var post_id = $(this).find(XKit.extensions.timestamps.blog_link_class)[1].href.split("/").slice(-2)[0];
+				var blog_name = $(this).find(XKit.extensions.timestamps.blog_link_class)[1].href.split("/")[2].split(".")[0];
+			} catch (e) {
+				note = $(this).find(XKit.extensions.timestamps.reblog_headers_class).append(`<div class="xtimestamp">&nbsp;</div>`);
+				XKit.extensions.timestamps.show_failed(note);
+			}
+
+			var normal_html = `<div class="xkit_timestamp_${post_id} xtimestamp xtimestamp_loading">&nbsp;</div>`;
+			$(this).find(XKit.extensions.timestamps.reblog_headers_class).append(normal_html);
+			var note = $(`.xkit_timestamp_${post_id}`);
+
+			// XKit.interface.react.page_props doesn't appear to work for reblogs yet
+//			XKit.extensions.timestamps.react_fetch_timestamp(post_id, note);
+			XKit.extensions.timestamps.fetch_timestamp(post_id, blog_name, note);
+		});
+	},
+
+	react_fetch_timestamp: async function(post_id, date_element) {
+		if (this.fetch_from_cache(post_id, date_element)) {
+			return;
+		}
```

Comment:
the cache is unnecessary now that timestamps are included in the post props - there's no additional network requests to save on when getting them

Reply:
if we're still using `fetch_timestamp` for reblog timestamps then this is indeed unnecessary

Reply:
Wait, why shouldn't we fetch from cache when using the regular fetch_timestamp for reblogs?

Reply:
we should, but this is in `react_fetch_timestamp`. [see this conversation in the Files tab](https://github.com/new-xkit/XKit/pull/1844/files/3c6fe29c8d9a43f17afeaf21285d84c357b4b9f3#diff-13325f4522cfe04acaed335ee61812f9R335)